numpy/testing/pytest_tools/noseclasses.py

```diff
@@ -0,0 +1,342 @@
+# These classes implement a doctest runner plugin for nose, a "known failure"
+# error class, and a customized TestProgram for NumPy.
+
+# Because this module imports nose directly, it should not
+# be used except by nosetester.py to avoid a general NumPy
+# dependency on nose.
+from __future__ import division, absolute_import, print_function
+
+import os
+import doctest
+import inspect
+
+import numpy
+import pytest
+from .utils import KnownFailureException, SkipTest
+import _pytest.runner
+import _pytest.skipping
+
+
+class NpyPlugin(object):
+    
+    def pytest_runtest_makereport(self, call):
+        if call.excinfo:
+            if call.excinfo.errisinstance(KnownFailureException):
+                #let's substitute the excinfo with a pytest.xfail one
+                call2 = call.__class__(
+                    lambda: _pytest.runner.skip(str(call.excinfo.value)),
+                    call.when)
+                print()
+                print()
+                print(call.excinfo._getreprcrash())
+                print()
+                print(call.excinfo)
+                print()
+                print(call2.excinfo)
+                print()
+                call.excinfo = call2.excinfo
+            if call.excinfo.errisinstance(SkipTest):
+                #let's substitute the excinfo with a pytest.skip one
+                call2 = call.__class__(
+                    lambda: _pytest.runner.skip(str(call.excinfo.value)),
+                    call.when)
+                call.excinfo = call2.excinfo
+
+
+if False:
```

Comment:
Remove this code?

Reply:
I guess the doctester isn't run? Should probably fix that.

Reply:
Update: I'm not going to fix the nosetester in this PR, but I also don't think it should be removed because it will eventually be needed.